### PR TITLE
Add and handle REST V1 (Other URL) messages

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/71337

Builds on: https://github.com/moderntribe/tribe-common/pull/313

This PR makes sure REST V1 messages are handled and used in 2 moments:

1. when the EA service answers the request and the response is packaged (especially errors)
2. when data is handled on the JS side

### CODE REVIEW

**Comment by lucatume** [comment]
@bordoni have a look at this as the [EA_Messages](https://github.com/moderntribe/the-events-calendar/blob/71ee7a4780f84ddaaaa9697d14e9b50b929e1929/src/Tribe/REST/V1/EA_Messages.php#L3-L2) class is probably of interest to you.
Essentially if EA returns messages in the format:

```
rest-v1::<slug>
```

Then that message will be reflected on the client with useful information like "TEC REST API is disabled on the target server" and so on.

File: src/Tribe/Aggregator/Service.php [hunk 1]
```diff
@@ -505,4 +480,44 @@ public function get_limit_remaining( $ignore_cache = false ) {
...
 	}
+
+	/**
+	 * Registers the message map used to translate message slugs returned from EA service into localized strings.
+	 *
+	 * These messages are delivered by the EA service and don't need to be registered. They just need to exist
+	 * here so that they can be translated.
+	 */
+	protected function register_messages()
+	{
```
Comment (bordoni) [inline src/Tribe/Aggregator/Service.php]: Don't break line here.

File: src/Tribe/Aggregator/Tabs/New.php [hunk 2]
```diff
@@ -456,6 +456,10 @@ public function ajax_fetch_import() {
 
 		$result = $record->get_import_data();
 
+		if ( isset( $result->data ) ){
```
Comment (bordoni) [inline src/Tribe/Aggregator/Tabs/New.php]: Spacing `){`

File: src/Tribe/REST/V1/Main.php [hunk 3]
```diff
@@ -167,4 +169,42 @@ public function get_reference_url() {
...
+	}
+
+	/**
+	 * Filters the messages returned by the Event Aggregator Service to add those specific to the REST API v1.
+	 *
+	 * @param array $messages
+	 * @return array The original messages plus those specific to the REST API V1.
+	 */
+	public function filter_service_messages( array $messages = array() )
+	{
```
Comment (bordoni) [inline src/Tribe/REST/V1/Main.php]: Dont break line

File: src/Tribe/REST/V1/Main.php [hunk 4]
```diff
@@ -167,4 +169,42 @@ public function get_reference_url() {
 	protected function url_prefix() {
 		return $this->url_prefix;
 	}
+
+	protected function hook_messages()
+	{
```
Comment (bordoni) [inline src/Tribe/REST/V1/Main.php]: Dont break line

File: src/Tribe/REST/V1/Messages.php [hunk 5]
```diff
@@ -35,4 +40,30 @@ public function get_message( $message_slug ) {
...
 		return '';
 	}
+
+	/**
+	 * Returns the associative array of all the messages handled by the class.
+	 *
+	 * @return array An associative array in the `[ <slug> => <localized message> ]` format.
+	 */
+	public function get_messages(  )
+	{
```
Comment (bordoni) [inline src/Tribe/REST/V1/Messages.php]: Don't break line

File: src/resources/js/aggregator-fields.js [hunk 6]
```diff
@@ -369,8 +369,12 @@ tribe_aggregator.fields = {
 			import_type = $( '#' + $import_type.first().attr( 'id' ).replace( 's2id_', '' ) ).val();
 		}
 
-		if ( 'manual' === import_type && ! data.items.length ) {
-			obj.display_fetch_error( ea.l10n.no_results );
+		if ('manual' === import_type && !data.items.length) {
```
Comment (bordoni) [inline src/resources/js/aggregator-fields.js]: Spacing.

File: src/resources/js/aggregator-fields.js [hunk 7]
```diff
@@ -369,8 +369,12 @@ tribe_aggregator.fields = {
...
+			var origin = data.origin;
+			var message = 'undefined' !== ea.l10n[origin].no_results ?
+				ea.l10n[origin].no_results
```
Comment (bordoni) [inline src/resources/js/aggregator-fields.js]: spacing.

File: src/Tribe/REST/V1/Messages.php [hunk 8]
```diff
@@ -35,4 +40,30 @@ public function get_message( $message_slug ) {
...
+	 *
+	 * Used to uniform the slug format to the one used by the `Tribe__Events__Aggregator__Service` class.
+	 *
+	 * @see Tribe__Events__Aggregator__Service::register_messages()
+	 *
+	 * @param string $message_slug
+	 *
+	 * @return string The prefixed message slug.
+	 */
+	public function prefix_message_slug( $message_slug )
```
Comment (bordoni) [inline src/Tribe/REST/V1/Messages.php]: dont break line

**Comment by lucatume** [comment]
Dracaris.
How difficult can it be to store code style settings on a per-project base and used them?!

Meh... this should go

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Comma required after last value in array declaration | [Main.php:190](https://github.com/moderntribe/the-events-calendar/blob/f5d8eac0e66fe722d5ee4b0dc857edfa4a7f4067/src/Tribe/REST/V1/Main.php#L190)

> via jenkins/codesniffer-phpcs

**Pull request merged**